scripts/publish-npm.js

```diff
@@ -46,133 +46,133 @@ const fs = require('fs');
+  // For non-nightly, non-dry-run, CircleCI job `prepare_package_for_release` does this
+  if (buildType === 'nightly' || buildType === 'dry-run') {
+    if (
+      exec(
+        `node scripts/set-rn-version.js --to-version ${version} --build-type ${buildType}`,
+      ).code
+    ) {
+      echo(`Failed to set version number to ${version}`);
+      return exit(1);
+    }
+  }
 
-// Write version number to the build folder
-const releaseVersionFile = path.join('build', '.version');
-fs.writeFileSync(releaseVersionFile, releaseVersion);
+  generateAndroidArtifacts(version);
 
-if (dryRunBuild) {
-  echo('Skipping `npm publish` because --dry-run is set.');
-  exit(0);
+  // Write version number to the build folder
+  const versionFile = path.join('build', '.version');
+  fs.writeFileSync(versionFile, version);
+
+  if (buildType === 'dry-run') {
+    echo('Skipping `npm publish` because --dry-run is set.');
+    return exit(0);
+  }
+
+  // We first publish on Maven Central all the necessary artifacts.
+  // NPM publishing is done just after.
+  publishAndroidArtifactsToMaven(version, buildType === 'nightly');
+
+  const tagFlag = `--tag ${tag}`;
+  const otp = process.env.NPM_CONFIG_OTP;
+  const otpFlag = otp ? ` --otp ${otp}` : '';
+
+  if (exec(`npm publish ${tagFlag}${otpFlag}`, {cwd: RN_PACKAGE_DIR}).code) {
+    echo('Failed to publish package to npm');
+    return exit(1);
+  } else {
+    echo(`Published to npm ${version}`);
+    return exit(0);
+  }
 }
 
-// Running to see if this commit has been git tagged as `latest`
-const isLatest = exitIfNotOnGit(
-  () => isTaggedLatest(currentCommit),
-  'Not in git. We do not want to publish anything',
-);
-
-// We first publish on Maven Central all the necessary artifacts.
-// NPM publishing is done just after.
-publishAndroidArtifactsToMaven(releaseVersion, nightlyBuild);
-
-const releaseBranch = `${major}.${minor}-stable`;
-
-// Set the right tag for nightly and prerelease builds
-// If a release is not git-tagged as `latest` we use `releaseBranch` to prevent
-// npm from overriding the current `latest` version tag, which it will do if no tag is set.
-const tagFlag = nightlyBuild
-  ? '--tag nightly'
-  : prerelease != null
-  ? '--tag next'
-  : isLatest
-  ? '--tag latest'
-  : `--tag ${releaseBranch}`;
-
-// use otp from envvars if available
-const otpFlag = otp ? `--otp ${otp}` : '';
-
-if (exec(`npm publish ${tagFlag} ${otpFlag}`, {cwd: RN_PACKAGE_DIR}).code) {
-  echo('Failed to publish package to npm');
-  exit(1);
-} else {
-  echo(`Published to npm ${releaseVersion}`);
-  exit(0);
+if (require.main === module) {
+  const argv = yargs
+    .option('n', {
+      alias: 'nightly',
+      type: 'boolean',
+      default: false,
+    })
+    .option('d', {
+      alias: 'dry-run',
+      type: 'boolean',
+      default: false,
+    })
+    .option('r', {
+      alias: 'release',
+      type: 'boolean',
+      default: false,
+    })
+    .strict().argv;
```

Comment:
Not sure if there's a way fix this, but something I found weird when reverse engineering this script and others was I initially thought one could multiple arguments, rather than just one of these (e.g: I thought you could pass both `dry-run` and `nightly`, or `dry-run` and `release`). The format allows this, so when I did so I just got weird errors and it was a while before I realized the builds are mutually exclusive 😅

Reply:
Yea that's a good point, these should be mutually exclusive. We could update the args here to enforce that but I didn't want to touch any CircleCI stuff in this PR. 
Maybe can follow up on that

Reply:
Yeah not a blocking comment or anything, just wanted to mention it